### PR TITLE
Add more documentation to fluent-bundle

### PR DESCRIPTION
**Pull request by gregtatum**
I wrote more docs as a way to learn the `fluent-bundle` codebase.

### CODE REVIEW

File: fluent-bundle/src/args.rs [hunk 1]
```diff
@@ -48,14 +56,17 @@ use crate::types::FluentValue;
 pub struct FluentArgs<'args>(Vec<(Cow<'args, str>, FluentValue<'args>)>);
 
 impl<'args> FluentArgs<'args> {
+    /// Creates a new empty list.
```
Comment (dminor) [inline fluent-bundle/src/args.rs:59]:
```suggestion
    /// Creates a new empty argument map.
```

File: fluent-bundle/src/resolver/mod.rs [hunk 2]
```diff
@@ -25,7 +31,9 @@ pub(crate) trait ResolveValue {
         M: MemoizerKind;
 }
 
+/// Resolves and AST node to a string that is written to source `W`.
```
Comment (dminor) [inline fluent-bundle/src/resolver/mod.rs:34]:
```suggestion
/// Resolves an AST node to a string that is written to source `W`.
```

File: fluent-bundle/src/resolver/mod.rs [hunk 3]
```diff
@@ -25,7 +31,9 @@ pub(crate) trait ResolveValue {
...
 pub(crate) trait WriteValue {
+    /// Resolves and AST node to a string that is written to source `W`.
```
Comment (dminor) [inline fluent-bundle/src/resolver/mod.rs:36]:
```suggestion
    /// Resolves an AST node to a string that is written to source `W`.
```

**Comment by gregtatum** [comment]
Rebasing due to conflicts.

**Pull request merged**